### PR TITLE
Create backend and frontend tests, refactor backend

### PR DESCRIPTION
**Pull request by StengoS**
Created backend tests with pytest, started development of frontend test with cypress, refactor backend functions (moved processing logic in backend to separate from any web/external calls)

### CODE REVIEW

File: myapp/backend/backend.py [hunk 1]
```diff
@@ -220,6 +202,30 @@ def get_team_roster():
...
+
+    if status is not None:
+        filters['member_status'] = status
+    if position is not None:
+        filters['member_position'] = position
+    if specialization is not None:
+        filters['member_specialization'] = {'$in': [f'{specialization}']}
+
+    return filters, single_name
+
```
Comment (tammychau99) [inline myapp/backend/backend.py:227]: I like how you split it up into its own function! I really like that and now it's more readable!

**Review by tammychau99 — APPROVED** [review]
Looks great! I love how you split up the filter by name functionality into its own function. Definitely more readable now!

File: myapp/cypress/support/commands.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// ***********************************************
```
Comment (jenisa10) [inline myapp/cypress/support/commands.js:1]: Why are all of these commented out?

File: myapp/cypress/integration/clubtracker_frontend_tests.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// Feature: Get to team roster page and sort by name
```
Comment (jenisa10) [inline myapp/cypress/integration/clubtracker_frontend_tests.js:1]: I love the story-based testing!

File: myapp/backend/backend_test.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
+import pytest
```
Comment (jenisa10) [inline myapp/backend/backend_test.py:1]: The tests seem quite extensive. Good job!

File: myapp/backend/backend.py (hunk 1 above)
Comment (jenisa10) [reply myapp/backend/backend.py:227]: Agreed!

**Review by jenisa10 — APPROVED** [review]
Hello Steven! Great job getting initial tests and everything. I think you did a great job making everything clean. I think something we can do onwards is just made comments on what each file is used for, as seeing it for the first time, I was quite confused, especially with the xml or iml files. I think you did great in creating a story-based method to work out the frontend testing and the backend was quite thorough. Great job!

**Pull request merged**